planner/funcdep/fd_graph.go

```diff
@@ -549,6 +549,280 @@ func (s *FDSet) MakeCartesianProduct(rhs *FDSet) {
+//			Pos eg: if the filter is on EFD's `from` cols, it's ok. Let's say: (a,b) * (c,d,e) on (a=c and a=2), a=2 only won't leading
+//			same key a with matched c and mismatched NULL, neg case result is changed as above, so strict FD {a} -> {c} can exist.
+//			a  b  |  c     d     e
+//			------+----------------
+//		 	1  1  |  1    NULL   1
+//		    1  2  | NULL  NULL  NULL
+//			2  1  | NULL  NULL  NULL
+//			Neg eg: left join with (a,b) * (c,d,e) on (a=c and b=c), two EFD here, where b=c can also refuse some rows joined by a=c,
+//			consequently applying it with NULL as (1  2  | NULL  NULL  NULL), leading the same key a has different value 1/NULL. But
+//			macroscopically, we can combine {a,b} together as the strict FD's from side, so new FD {a,b} -> {c} is secured. For case
+//			of (a=c and b=ce), the FD is {a, b} -> {c, e}
+//
+// 			conclusion: without this kind of limited left conditions to judge the join match, we can say: FD {a} -> {c} exists.
+//
+//		<3.2> equivalence FD: when the determinant and dependencies from a equivalence FD of join condition are each covering a strict
+//			FD of the left / right side. The left side strict FD's dependencies can be extended to all cols after join result.
+//			a  b  |  c     d     e
+//			------+----------------
+//		 	1  1  |  1    NULL   1
+//		    2  2  | NULL  NULL  NULL
+//		    3  1  | NULL  NULL  NULL
+//			left join with (a,b) * (c,d,e) on (a=c and b=1). Supposing that left `a` are strict Key and right `c` are strict Key too.
+//			Key means the strict FD can determine all cols from that table.
+//			case 1: left join matched
+//				one left row match one / multi rows from right side, since `c` is strict determine all cols from right table, so
+//              {a} == {b} --> {all cols in right}, according to the transitive rule of strict FD, we get {a} --> {all cols in right}
+//			case 2: left join miss match
+//				miss matched rows from left side are unique originally, even appended with NULL value from right side, they are still
+//				strictly determine themselves and even the all rows after left join.
+//			conclusion combined:
+//				equivalence with both strict Key from right and left, the left side's strict FD can be extended to all rows after left join.
+//
+//		<3.3> equivalence FD: let's see equivalence FD as double-directed strict FD from join equal conditions, and we  only keep the
+//			rhs ~~> lhs.
+//			a  b  |  c     d     e
+//			------+----------------
+//		 	1  1  |  1    NULL   1
+//		    1  2  | NULL  NULL  NULL
+//		    2  1  | NULL  NULL  NULL
+//			left join with (a,b) * (c,d,e) on (a=c and b=1). From the join equivalence condition can derive a new FD {ac} == {ac}.
+//			while since there are some supplied null value in the c column, we don't guarantee {ac} == {ac} yet, so do {a} -> {c}
+//			because two same determinant key {1} can point to different dependency {1} & {NULL}. But in return, FD like {c} -> {a}
+//			are degraded to the corresponding lax one.
```

Comment:
These comments are too hard to understand. Please make it grammatically correct at least.